File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 18/CS/ConnectionZoneAppCS/Consumer.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;

public partial class Consumer : System.Web.UI.UserControl
{
    [ConnectionConsumer("Text", "TextConsumer")]
    public void RetriveDataInterface(IDataTransmit sender)
    {
        Label2.Text = sender.RetriveData();
    }

}

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 18/VB/ConnectionZoneAppVB/Consumer.ascx.vb

Partial Class Consumer
    Inherits System.Web.UI.UserControl
    <WebControls.WebParts.ConnectionConsumer("Text", "TextConsumer")> _
    Public Sub RetriveDataInterface(ByVal sender As IDataTransmit)
        Label2.Text = sender.RetriveData()
    End Sub
End Class
